### ISSUE TITLE
[Rust] Implement new Concept Exercise: string-vs-str

### ISSUE THREAD

**Issue by efx** [exercism/v3#1095]
This issue describes how to implement the `string-vs-str` concept exercise for the rust track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

Teach the student the differences between `String` and `&str` and how to work successfully with both.

## Learning objectives

- Know `String` is owned and mutable
- Know `&str` is shared and immutable
- Know `String` has `&str` methods because of shared traits 
- Know of performance characteristics

## Out of scope

- Machine level representation of either
- in depth performance differences

## Concepts

- mutability
- ownership
- slices

## Prerequisites

- basics
- slices
- str-use
- string-use

### Hints

- [string slices discussion in TRPL book][https://doc.rust-lang.org/book/ch04-03-slices.html#string-slices]
- [`&str` primitive documentation](https://doc.rust-lang.org/std/primitive.str.html)
- [`std::str`](https://doc.rust-lang.org/std/str/)
- [string types in RBE](https://doc.rust-lang.org/stable/rust-by-example/std/str.html)

### After

- [tutorial on string versus str in functions](http://hermanradtke.com/2015/05/03/string-vs-str-in-rust-functions.html)
- https://fasterthanli.me/blog/2020/working-with-strings-in-rust/
- https://doc.rust-lang.org/book/ch08-02-strings.html#storing-utf-8-encoded-text-with-strings

## Representer

- No changes needed

## Analyzer

- No changes needed

## Implementing

See our [implementation guide](https://github.com/exercism/v3/blob/master/languages/rust/howto/implementing-a-concept-exercise.md#how-to-implement-a-rust-concept-exercise) for guidance.
In addition you may want to make use of the shell script while in the concept directory, with the arg string-vs-str, e.g. `sh boil_exercise.sh string-vs-str`

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

**Comment by gilescope** [comment]
I'd like to take a crack at this.
(there could be a later exercise to go from OsStr / OsString through str/String to CStr / CString maybe. The relationships between those types and knowing how to do back and forth is good practice.)
